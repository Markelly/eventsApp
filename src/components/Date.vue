<template>
  <div class="date" v-text="getDate"></div>
</template>

<script>
import { monthName, weekDayName } from "../utils/constants";

export default {
  name: "Date",
  props: {
    date: {
      type: Object,
      required: true
    }
  },
  computed: {
    getDate() {
      let formattedDate = "";
      const date = new Date(
        this.date && this.date.start && this.date.start.localDate
      );
      if (date) {
        let weekday = weekDayName[date.getDay()];
        let month = monthName[date.getMonth()];
        let day = date.getDate();
        let year = date.getFullYear();
        formattedDate = `${weekday}, ${day} ${month} ${year}`;
      }
      return formattedDate;
    }
  }
};
</script>

<style scoped lang="scss">
@import "../styles/events-app";

.date {
  color: $color-primary;
  font-weight: 600;
  padding: 4px;
  font-size: 12px;

  @media (min-width: $breakpoint-tablet) {
    font-size: 16px;
    padding: 8px;
  }
}
</style>
