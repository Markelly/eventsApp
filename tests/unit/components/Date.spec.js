import Date from "@/components/Date";
import { createLocalVue, shallowMount } from "@vue/test-utils";

const localVue = createLocalVue();

describe("<Date/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Date, {
      localVue,
      propsData: {
        date: {
          start: {
            localDate: "2020-04-03"
          }
        }
      }
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should display the formatted date", () => {
    const formattedDate = "Friday, 3 April 2020";
    expect(wrapper.vm.getDate).toEqual(formattedDate);
  });
});
