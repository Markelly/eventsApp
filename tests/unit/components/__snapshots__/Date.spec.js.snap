// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Date/> should render the component 1`] = `<div class="date">Friday, 3 April 2020</div>`;
